cf/data/data.py

```diff
@@ -1239,26 +1248,64 @@ def __keepdims_indexing__(self):
     def __keepdims_indexing__(self, value):
         self._custom["__keepdims_indexing__"] = bool(value)
 
-    def _conform_after_dask_update(self):
-        """Remove elements made invalid by updating the `dask` array.
+    def _clear_after_dask_update(self, clear=_ALL):
+        """Remove components invalidated by updating the `dask` array.
 
         Removes or modifies components that can't be guaranteed to be
-        consistent with an updated `dask` array`:
-
-        * Deletes a source array.
-        * Deletes cached element values.
+        consistent with an updated `dask` array`. See the *clear*
+        parameter for details.
 
         .. versionadded:: 3.14.0
 
+        .. seealso:: `_del_Array`, `_del_cached_elements`, `_set_dask`
+
+        :Parameters:
+
+            clear: `int`, optional
+                Specify which components should be removed. Which
+                components are removed is determined by sequentially
+                combining *clear* with the ``_ARRAY`` and ``_CACHE``
+                `integer-valued contants, using the bitwise AND
+                operator:
+
+                * If ``clear & _ARRAY`` is non-zero then a source
+                  array is deleted.
+
+                * If ``clear & _CACHE`` is non-zero then cached
+                  element values are deleted.
+
+                By default *clear* is the ``_ALL`` integer-valued
+                constant, which results in all components being
+                removed.
+
+                If *clear* is the ``_NONE`` integer-valued constant
+                then no components are removed.
+
+                To retain a component and remove all others, use
+                ``_ALL`` with the bitwise OR operator. For instance,
+                if *clear* is ``_ALL ^ _CACHE`` then the cached
+                element values will be kept but all other components
+                will be removed.
+
+                .. versionadded:: 3.14.1
+
         :Returns:
 
             `None`
 
         """
-        self._del_Array(None)
-        self._del_cached_elements()
+        if not clear:
+            return
+
+        if clear & _ARRAY:
+            # Delete a source array
+            self._del_Array(None)
 
-    def _set_dask(self, array, copy=False, conform=True):
+        if clear & _CACHE:
+            # Delete cached element values
+            self._del_cached_elements()
+
+    def _set_dask(self, array, copy=False, clear=_ALL):
```

Comment:
The 'seealso' admonition for this method still references the old `_conform_after_dask_update` and needs updating (but a `grep` on `cf/` on this branch shows that that one is the last remaining reference to the old name).

Reply:
Fixed: https://github.com/NCAS-CMS/cf-python/pull/607/commits/64b028d94ec591a71c90c4efa16b9bfbb570ac55